cmd/boot-script-service/default_api.go

```diff
@@ -574,8 +574,8 @@ func paramSubstitute(params, pvar string, getVal paramValRetreiver) (string, err
 // BootData and additional parameters provided.  The resultant script is
 // returned as a string.  If an error occurs, a null string is returned along
 // with the error.
-func buildBootScript(bd BootData, sp scriptParams, chain, descr string) (string, error) {
-	debugf("buildBootScript(%v, %v, %v, %v)\n", bd, sp, chain, descr)
+func buildBootScript(bd BootData, sp scriptParams, chain, role string, subRole string, descr string) (string, error) {
```

Comment:
When you give a parameter without a type, it has the type of the next parameter. So chain is a string because descr is a string in the original code. I believe you could add role and subRole without a type as well, as they are all strings. Otherwise, I would probably add the string type to chain as well. Otherwise, this seems inconsistent.

Reply:
Thanks for pointing that out. I didn't know this about go. I was wondering why chain worked. The fix should be uploaded soon.